_content/devices/bytes.md

```diff
@@ -158,38 +166,73 @@ int myVal = ((int)(payload[0]) << 8)
 
 ### 4. Shift bits
 
-If the range of expected values is bigger than 65536 we can use the same trick. The only difference is that we have to manually shift bits when we encode on Arduino, just like we did in the payload function.
+If the range of expected values is bigger than 65536 we can use the same trick. The only difference is that we have to manually shift bits when we encode on Arduino, just like we did in the payload function, and we need to be careful while decoding in JavaScript.
 
-Let's say we need to encode a [long](https://www.arduino.cc/en/Reference/Long) which uses 4 bytes for a range up to 4294967296.
+Let's say we need to encode an unsigned [long](https://www.arduino.cc/en/Reference/Long) which uses 4 bytes for a range of 0 up to 4294967296.
 
 Encode (Arduino):
 
 ```c
-long lng = 200000L;
+unsigned long lng = 200000L;
 byte payload[4];
 payload[0] = (byte) ((lng & 0xFF000000) >> 24 );
 payload[1] = (byte) ((lng & 0x00FF0000) >> 16 );
 payload[2] = (byte) ((lng & 0x0000FF00) >> 8  );
 payload[3] = (byte) ((lng & 0X000000FF)       );
 ```
 
+For Arduino you will often see that the masks such as `0x00FF0000`, and the type casts to `(byte)`, are not used. That works when `payload` is defined as an array of `byte` as that will really only fit 8 bits, where only the 8 least significant bits are stored and all others are simply discarded. But it never hurts to be explicit, and other platforms might surely need the masks and casts.
```

Comment:
This is not entirely true; `(lng & 0x0000FF00) >> 8` is still an `unsigned long`. If we cast, we don't need the mask. I don't know why we added the mask here.

Reply:
I wonder if, for Arduino, one even needs the cast? Examples such as https://www.thethingsnetwork.org/forum/t/how-to-use-the-dht22-with-lmic/15708/4 (okay, that's not a long, but mapping a 16 bits integer to an 8 bits array) always worked for me:

```c
int16_t temp = 10 * dht.readTemperature(true);

uint8_t buff[3]; // reserve 3 bytes in memory

// Handle high byte (MSB) first; 0xFF for -234 or 0x00 for +234
// 0xFF16 >> 8 shifts the 0x16 out of memory, leaving 0x00FF
// 0x00FF does not fit in a single byte, so only 0xFF is stored in buff[0]:
buff[0] = temp >> 8;

// Handle low byte (LSB) next; 0x16 for -234 or 0xEA for +234
// 0xFF16 does not fit in a single byte, so only 0x16 is stored in buff[1]:
buff[1] = temp;
```

Reply:
As far as I know it is not necessary to mask the source as it is explicitly stored in a byte. If a mask is applied it is normally done after shifting.

Ex:
`buff[0] = (temp >> 16) & 0xFF;`

Reply:
@jpmeijers here the mask isn't necessary either